spec/spec.md

```diff
@@ -101,6 +116,35 @@ After evaluating the arguments (if any), control is passed to the callee.
 
 > TODO: specification for the built-in operations is missing
 
+#### Type Expression
+
+```grammar
+type_expr ::= (UnitTy)  # unit
+           |  (BoolTy)  # bool
+           |  (IntTy)   # int
+           |  (FloatTy) # float
+```
+
+### Declarations
+
+#### Procedure
+
+```grammar
+decl ::= (ProcDecl name:<ident> ret:<type_expr> params:(Params) body:<expr>)
+```
+
+Let `S` be the current scope. If `lookup(S, name)` succeeds, an error is
+reported. Otherwise, `name` is added to `S`, referring to the declared
+procedure.
+
+`body` must be of type `void`. It is the computation that is run when the
+procedure is called.
```

Comment:
Ah! So if we want to add expression based bodies, then we add in a language layer that does automatic `return <value>` insertion.

Reply:
Yep, that's the idea.